File: android/feature/explore/src/main/java/com/enterprise/feature/explore/domain/repositories/ExploreFeedRepository.kt
package com.enterprise.feature.explore.domain.repositories

import com.enterprise.core.domain.entities.DomainResult
import com.enterprise.core.domain.entities.UIComponent
import com.enterprise.core.domain.repositories.FeedRepository
import kotlinx.coroutines.flow.Flow

/** Extended feed repository with personalization and search. */
interface ExploreFeedRepository : FeedRepository {
    fun getPersonalizedFeed(userId: String): Flow<DomainResult<List<UIComponent>>>
    suspend fun searchFeed(query: String): DomainResult<List<UIComponent>>
}

File: ios/EnterpriseMobileApp/Sources/Feature/Explore/Domain/Repositories/ExploreFeedRepository.swift
import Foundation
import CoreDomain

/// Feature-specific repository for Explore feed operations.
public protocol ExploreFeedRepository: FeedRepository {
    func getPersonalizedFeed(userId: String) -> AsyncStream<DomainResult<[UIComponent]>>
    func searchFeed(query: String) async -> DomainResult<[UIComponent]>
}
